### ISSUE TITLE
Single Page Application and Turbolinks

### ISSUE THREAD

**Issue by lucadegasperi** [turbolinks/turbolinks#300]
Hello, not sure wether I should post this in one of the adapters repo or here. I'm currently working on an SPA with Vue.js and wanted to embed the app inside one of the Turbolinks adapters in order to get the native view transitions for free. Now it seems overkill to redraw the whole body of the page since the SPA takes care of the application state and routing. Is there a way to prevent the body replacement that turbolinks does but still get the native view transitions to trigger?

Basically I would like to skip the body replacement part (which is useless in an SPA) but still get the native transitions.

**Comment by murb** [comment]
I'm afraid you're trying to use a hammer to drive a screw. You should catch the javascript events when transitioning in your native app, but I'm that it is beyond something to discuss on the Turbolinks issues page (although maybe you can find some inspiration in the turbolinks-ios / android code?

**Comment by domchristie** [comment]
> not sure wether I should post this in one of the adapters repo or here.
> Is there a way to prevent the body replacement that turbolinks does but still get the native view transitions to trigger?

Yes, this is more related to the native adapters. Replacing the body is fundamental to Turbolinks in the browser, however, there might be a solution somewhere…

I'm not familiar with how/when Vue components are initialised or how they are rendered, so I am not sure whether this will work, but one approach might be:

- Browser client: do not include Turbolinks at all, use Vue router to control navigation
- Native client (i.e. turbolinks-ios/android): do not use Vue router, use Turbolinks to control navigation

This might require a bit of thought to maintain and serve up different JavaScript packages depending on the context, but it might work! Hope that helps.
